libsolidity/formal/SMTEncoder.h

```diff
@@ -313,8 +313,6 @@ class SMTEncoder: public ASTConstVisitor
 
 	smt::VariableUsage m_variableUsage;
 	bool m_arrayAssignmentHappened = false;
-	// True if the "No SMT solver available" warning was already created.
-	bool m_noSolverWarning = false;
```

Comment:
Why did you need to move this from here?

Reply:
Because now BMC and CHC each need one separately

Reply:
I get that, but making it protected in `SMTEncoder` should work?
I mean, instances of CHC and BMC will have it's own value for it. Or I am missing something

Reply:
Right, now I don't know actually. I thought they could conflict somehow but that doesn't make sense, will move it back